### DISCUSSION TITLE
How to reset imported form elements?

### DISCUSSION THREAD

**Discussion by ghost** [neuefische/questions#632] [Web] [answered]
In my Component "new-pedal-form" there are most input elements created directly, but two are imported from their own component ("form-tag-input" and "form-category-input"). In "new-pedal-form" there are two buttons, one to submit (console.log()) and reset the form, and one to cancel and reset the form. When clicking one of these buttons, the input fields are cleared, except the imported ones.) I haven't found about this in this board neither in stack overflow. Maybe i ask the wrong question...

[the branch:](https://github.com/JonasDaJones/capstone-project/tree/pedal-form-frirst)
Any idea is appreciated
Thank you,
Jonas

**Comment by actyvystom** [comment]
Hi @JonasDaJones, your imported form elements manage their own (local) state for the input values. 
If you lift up the state to your parent form component, the reset should work.
Cheers, Thomas

**Reply by ghost** [reply]
thank you, (maybe i made some mistakes when trying this out. I will try the lift up more patiently and accurately now).

**Comment by julialoeschel** [comment, marked as answer]
Hey Jonas. 
You are holding a state `addedTags` which is not really necessary. You could use the `tags`instead.
` <StyledTagView>Tags: {tags.join(", ")}</StyledTagView>`

and ` <TagInput onSaveTag={handleTagSave} tags={tags} />` in NewPedalForm.
